Make the movie lister example work on Windows

jpvandervelden opened the pull request:
As a Java guy I'd like to use the dependency injection in Python too. I found this library. It didn't work for me at first on Windows & VS Code. I made some changes so now it works on Linux and Windows.

Keep up the good work!

jpvandervelden committed:
Create a bash script and batch file to do similar things on WSL and Windows.

jpvandervelden committed:
- Changed the creation of the CSV file so it doesn't add an additional newline, which made the example fail on Windows.
- Changed the relative imports to absolute imports so the example works out of the box in VS Code.
- Added a shell script and a batch file to run the example on Linux and Windows.

rmk135 reviewed (commented):
Hey @jpvandervelden :) Thanks a lot for the contribution and glad you've figured it out how to run it on windows. I posted a couple of actions. Appreciate if you could take a look.

jpvandervelden commented on examples/miniapps/movie-lister/setup_and_run_movie_lister.bat:
```diff
@@ -0,0 +1,11 @@
+echo on
+set PYTHON37=%UserProfile%\AppData\Local\Programs\Python\Python37\python.exe
```
This line could be removed and a comment added that the assumption is made that PYTHONPATH is pointing to the python installation.

jpvandervelden commented on examples/miniapps/movie-lister/setup_and_run_movie_lister.sh:
```diff
@@ -0,0 +1,14 @@
+
+rm -rf wslenv2
+
+python3 -m venv wslenv2
```
Correct.

jpvandervelden commented on examples/miniapps/movie-lister/data/fixtures.py:
```diff
@@ -18,10 +18,9 @@
 
 def create_csv(movies_data, path):
-    with open(path, 'w') as opened_file:
+    with open(path, 'w', newline='') as opened_file:
```
The error message was:
`TypeError: __init__() missing 3 required positional arguments: 'title', 'year', and 'director'`
which was caused by the CSV data being created with extra carriage return and new line.

The movies.csv file was like this:
```
The Hunger Games: Mockingjay - Part 2,2015,Francis Lawrence

Rogue One: A Star Wars Story,2016,Gareth Edwards

The Jungle Book,2016,Jon Favreau
```

ZipFile replied:
This was a valid issue with our example. According to the [csv module docs](https://docs.python.org/3/library/csv.html), files should be opened with `newline=''`. Either way, this fix is already implemented in our codebase.

jpvandervelden commented on examples/miniapps/movie-lister/setup_and_run_movie_lister.bat:
```diff
@@ -0,0 +1,11 @@
+
+%PYTHON37% -m venv %ENV_NAME%
+
+Start /WAIT cmd /k "%VENV_DIR%\activate & %PYTHON_VENV% -m pip install --upgrade pip & %PYTHON_VENV% -m pip install -r .\requirements.txt & set MOVIE_FINDER_TYPE=csv& %PYTHON_VENV% -m movies & set MOVIE_FINDER_TYPE=sqlite& %PYTHON_VENV% -m movies&%VENV_DIR%\deactivate & pause & exit 0"
```
Yes.. it is,  by adding a ^as a multiline command separator.
I will check in the file again.

jpvandervelden committed:
Removed the reference to the local python installation.
Changed script to use multiline command separator for the long command sequence.
Added some comment lines.

jpvandervelden commented on .gitignore:
```diff
@@ -72,3 +72,12 @@ src/dependency_injector/providers/*.so
 
 # Workspace for samples
 .workspace/
+/examples/miniapps/movie-lister/wslenv2-win/Scripts
```
This is how the virtual environment is created under Windows.

jpvandervelden commented on examples/miniapps/movie-lister/movies/__main__.py:
```diff
@@ -4,8 +4,8 @@
 
-from .listers import MovieLister
-from .containers import Container
+from movies.listers import MovieLister
```
It was a problem when using VS Code..
